lib/active_merchant/billing/gateways/decidir.rb

```diff
@@ -311,8 +316,14 @@ def message_from(success, response)
         message
       end
 
-      def success_from(response)
-        response['status'] == 'approved' || response['status'] == 'pre_approved'
+      def success_from(response, options)
+        status = %w(approved pre_approved)
+
+        if options[:action] == 'inquire'
+          status.include?(response['status']) || response['status'] == 'rejected'
```

Comment:
LGTM! This makes sense, given we would consider a 'rejected' returned status as a successful inquiry result.